drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/clipboard/Clipboard.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.dtable.client.editor.clipboard;
+
+import java.util.Set;
+
+import org.drools.workbench.models.guided.dtable.shared.model.DTCellValue52;
+
+public interface Clipboard {
+
+    /**
+     * Set data on clipboard
+     * @param data Selected cells and their original coordinates.
+     */
+    void setData( final Set<ClipboardData> data );
+
+    /**
+     * Get the data on the clipboard. Could be null.
+     * @return
+     */
+    Set<ClipboardData> getData();
+
+    /**
+     * Does the clipboard contain (non-null) data.
+     * @return
+     */
+    boolean hasData();
+
+    void clear();
+
+    /**
+     * A single cell
+     */
+    interface ClipboardData {
+
+        /**
+         * Source Row index.
```

Comment:
Just a minor detail here:
From what I understand from the code, the (row | column) index is relative to the cell selection (5th row in the selection, but 12th in the whole table), not absolute (like 5th row in the whole table, 2nd in the selection). In that case, it might be good to also clarify this in the javadoc.